apisix/init.lua

```diff
@@ -868,6 +869,54 @@ local function healthcheck_passive(api_ctx)
 end
 
 
+function _M.status()
+    core.response.exit(200, core.json.encode({
+        status = "ok" }),
+        { ["Content-Type"] = "application/json"
+    })
+end
+
+function _M.status_ready()
+    local local_conf = core.config.local_conf()
+    local role = core.table.try_read_attr(local_conf, "deployment", "role")
+    local provider = core.table.try_read_attr(local_conf, "deployment", "role_" .. role, "config_provider")
+    if provider == "yaml" or provider == "etcd" then
+        local status_shdict = ngx.shared["status-report"]
+        local ids = status_shdict:get_keys()
+        local errors = {}
+
```

Comment:
If the `ids` length is not equal to `ngx.worker.count`, it should also be reported as unhealthy.
Note that worker ID start at 0.